plugins/woocommerce/tests/php/includes/class-wc-tracks-test.php

```diff
@@ -210,4 +211,60 @@ public function test_store_id_is_added_to_properties() {
 		$this->assertEquals( $store_id, $properties['store_id'] );
 		delete_option( \WC_Install::STORE_ID_OPTION );
 	}
+
+	/**
+	 * Test that get_blog_details ensures the store ID is set.
+	 */
+	public function test_get_blog_details_ensures_store_id_is_set() {
+		// Delete the store ID option to simulate a fresh installation.
+		delete_option( \WC_Install::STORE_ID_OPTION );
+		delete_transient( 'wc_tracks_blog_details' );
+
+		// Call get_blog_details which should ensure store ID is set.
+		$blog_details = \WC_Tracks::get_blog_details( get_current_user_id() );
+
+		// Verify that store_id exists and is not null.
+		$this->assertArrayHasKey( 'store_id', $blog_details );
+		$this->assertNotNull( $blog_details['store_id'] );
+
+		// Verify that the store ID option was actually set in the database.
+		$store_id = get_option( \WC_Install::STORE_ID_OPTION );
+		$this->assertNotEmpty( $store_id );
+		$this->assertEquals( $store_id, $blog_details['store_id'] );
+	}
+
+	/**
+	 * Test that get_blog_details uses cached data from transients.
+	 */
+	public function test_get_blog_details_uses_cached_data() {
+		// Delete existing transient to start fresh.
+		delete_transient( 'wc_tracks_blog_details' );
+
+		// Set a known store ID.
+		$test_store_id = 'test_store_id_' . time();
+		update_option( \WC_Install::STORE_ID_OPTION, $test_store_id );
+
+		// First call should set the transient.
+		$first_call = \WC_Tracks::get_blog_details( get_current_user_id() );
+		$this->assertEquals( $test_store_id, $first_call['store_id'] );
+
+		// Change the store ID in the database.
+		$new_store_id = 'new_store_id_' . time();
```

Comment:
Nit: `time()` here has a chance to generate the same value if execution is fast enough. I think using uniqid might be better

```suggestion
		$test_store_id = 'test_store_id_' . uniqid();
		update_option( \WC_Install::STORE_ID_OPTION, $test_store_id );

		// First call should set the transient.
		$first_call = \WC_Tracks::get_blog_details( get_current_user_id() );
		$this->assertEquals( $test_store_id, $first_call['store_id'] );

		// Change the store ID in the database.
		$new_store_id = 'new_store_id_' . uniqid();
```